MyExample/FormSkiCompetition.cs

```diff
@@ -78,22 +104,34 @@ public void RefreshGrid()
 
         private void dataGridViewCompetitors_KeyDown(object sender, KeyEventArgs e)
         {
-            if(e.KeyCode == Keys.Delete)
+            if (e.KeyCode == Keys.Delete)
             {
                 int selected = Convert.ToInt32(dataGridViewCompetitors.SelectedRows[0].Cells[0].Value);
                 dataProvider.Delete(selected);
                 RefreshGrid();
             }
         }
 
-        private void VitoshaCompetition_Click(object sender, EventArgs e)
+        private void BigFinal_Click(object sender, EventArgs e)
         {
-            var fr = new FormRank(_connection);
+            var csf = new CompetitionSelectForm(_connection);
+            csf.Show();
+        }
 
-            fr.ShowDialog();
-            
-            dataGridViewMaleAvg.DataSource = dataProvider.AverageTimeMale();
-            dataGridViewFemaleAvg.DataSource = dataProvider.AverageTimeFemale();
+        private void listBox1_DoubleClick(object sender, EventArgs e)
+        {
+            if (listBox1.SelectedItem != null)
+            {
+                ListBox();
+                WaitForm wf = new WaitForm(_connection);
+                wf.Show(this);

```

Comment:
Initialize FormRank here, after the wait finishes. (Look at the other comment in WaitForm)